flink-ml-lib/src/test/java/org/apache/flink/ml/classification/OnlineModelSaveLoadTest.java

```diff
@@ -0,0 +1,128 @@
+                    .column("f1", DataTypes.of(DenseVector.class))
+                    .build();
+
+    private static final List<Row> modelData =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of(Vectors.dense(2.0, 4.5, 3.0)),
+                            Row.of(Vectors.dense(2.1, 4.6, 3.1)),
+                            Row.of(Vectors.dense(20.1, 5.6, 3.1)),
+                            Row.of(Vectors.dense(2.1, 4.7, 3.1))));
+
+    private static final List<Row> validData =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of(1.0, Vectors.dense(1.0, 3.5, -4.0)),
+                            Row.of(0.0, Vectors.dense(1.1, -8.6, 3.3))));
+
+    @Test
+    public void saveAndLoadOnlineModel() throws Exception {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+        env.setParallelism(1);
+
+        String tmpPath = tempFolder.newFolder().getAbsolutePath();
+
+        /* Constructs online LogisticRegression model. */
+        Table lrModelStream =
+                tEnv.fromDataStream(env.fromCollection(modelData), modelSchema).as("coefficient");
+        LogisticRegressionModel lr = new LogisticRegressionModel();
+        /* Saves online model to given path (tmpPath). */
+        lr.setModelData(lrModelStream);
+        lr.save(tmpPath);
+        env.execute();
+
+        /* Constructs validated data table. */
+        Table validDataTable =
+                tEnv.fromDataStream(env.fromCollection(validData), dataSchema).as("label, vec");
+
+        Process proc = Runtime.getRuntime().exec("ls " + tmpPath + "/data");
+        BufferedReader bufferedReader =
+                new BufferedReader(new InputStreamReader(proc.getInputStream()));
+
+        /* Loads every LogisticRegression model in model path and validates it. */
+        String modelVersion;
+        while ((modelVersion = bufferedReader.readLine()) != null) {
+            if (!"metadata".equals(modelVersion)) {
+                LogisticRegressionModel lrModel =
+                        new LogisticRegressionModel().setFeaturesCol("vec");
+                Table modelData =
+                        tEnv.fromDataStream(
+                                        loadModelData(
+                                                env,
+                                                tmpPath,
+                                                new LogisticRegressionModelData.ModelDataDecoder(),
+                                                modelVersion))
+                                .as("label, vec");
```

Comment:
Why do you convert model data as `label, vec`?

Reply:
The validated data has feature(vec) and label. I will check the prediction result  agree with given label.